src/main/kotlin/coverosR3z/server/utility/ServerUtilities.kt

```diff
@@ -114,6 +115,13 @@ class ServerUtilities() {
                 // without getting anything.  See SocketWrapper and soTimeout
                 logger.logTrace { "read timed out" }
                 shouldKeepAlive = false
+            } catch (ex : SocketException) {
+                if (ex.message?.contains("Connection reset") == true) {
+                    logger.logTrace { "client closed their connection while we were waiting to read from the socket" }
```

Comment:
more properly handling the case when a client shuts down their connection to us, we'll pass this up to be handled closer to the top loop of the server